### ISSUE TITLE
sc2: SC2-compatible LSTM policy with available-actions masking and temporal memory

### ISSUE THREAD

**Issue by espenhk** [espenhk/gamer-ai#109]
The existing `LSTMEvolutionPolicy` (`framework/policies.py`) uses CMA-ES-style evolutionary search over flattened LSTM weights. SC2 makes temporal memory especially valuable: army selection state persists across steps, the agent must track unscouted map regions, and rewards are delayed (win/loss only at episode end). An LSTM policy is the natural fit for these long-horizon dependencies.

### 1. Multi-head output layer

TMNF's LSTM emits logits over 9 discrete actions from a single output head. SC2 needs two independent heads:

```
LSTM hidden state (hidden_size,)
         │
    W_out: (hidden_size → 15)
         │
  ┌──────┴──────┐
  │             │
fn_logits    spatial_logits
  (6,)           (9,)
```

Output layer: `W_out: hidden_size → 15`, first 6 values = fn logits, last 9 = spatial logits.

### 2. Available-actions masking at inference

```python
fn_logits, spatial_logits = output[:6], output[6:]
fn_logits[unavailable_fn_mask] = -np.inf
fn_probs = softmax(fn_logits)
fn_idx = np.random.choice(6, p=fn_probs)
cell_idx = np.random.choice(9, p=softmax(spatial_logits))
```

### 3. Hidden state reset policy

Current code always resets `(h, c)` to zeros at episode start. For SC2 ladder it may be useful to carry hidden state across timeout-truncated resets within the same match. Expose a `reset_on_episode: bool` config flag (default `true`).

### 4. Larger hidden size and weight count

SC2 obs is 21-dim (ladder) vs 15-dim (TMNF). With `hidden_size: 64`:

```
LSTM input weights:  (obs_dim + hidden_size) × 4 × hidden_size
                   = (21 + 64) × 4 × 64 = 21,760
Output weights:      hidden_size × 15 = 960
Total θ ≈ 22,720 parameters
```

This is large but feasible for CMA-ES with `population_size: 20` over many generations, as each generation only evaluates fitness (no backprop).

## Recommended hyperparams (`games/sc2/config/training_params.yaml`)

```yaml
policy_type: sc2_lstm
policy_params:
  hidden_size: 64          # vs 32 for TMNF — larger obs/action space
  population_size: 20
  initial_sigma: 0.03      # smaller than CMA-ES linear: LSTM weights are sensitive
  reset_on_episode: true
```

## Tasks

- [ ] Implement `SC2LSTMPolicy` in `games/sc2/sc2_policies.py` with multi-head output (fn + spatial slices of `W_out`) and `reset_on_episode` flag
- [ ] Available-actions masking in `act()` using `info["available_fn_ids"]`
- [ ] CMA-ES weight evolution loop identical to `LSTMEvolutionPolicy` but with updated weight shapes and packing
- [ ] Champion serialisation to numpy `.npz` (same format as `LSTMEvolutionPolicy`)
- [ ] Register as `sc2_lstm` in the policy factory
- [ ] Tests: hidden state shape correct; output splits into fn (6) and spatial (9) logits; masking never selects unavailable fn; weight count matches formula; `reset_on_episode: false` carries `(h, c)` across resets

## Relevant files

- `framework/policies.py` — `LSTMEvolutionPolicy` base class
- `games/sc2/client.py` — `available_fn_ids` in info dict
- `games/sc2/actions.py` — grid cell → (x, y) coordinates
- New: `games/sc2/sc2_policies.py`
